File: backends/php-laravel/app/Domain/Client/Condition/ClientCondition.php
<?php
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
namespace App\Domain\Client\Condition;

use App\Domain\Client\Enums\ClientStatus;
use App\Support\Repositories\Conditions\AbstractCondition;
use Carbon\Carbon;

/**
 * クライアントConditionクラスです。
 *
 * @author Satoshi Nagashiba <nagashibas@sii-japan.co.jp>
 * @package App\Domain\Client\Condition
 */
class ClientCondition extends AbstractCondition
{    public ?string $name = null;
    public ?string $identifer = null;
    public ?string $postCode = null;
    public ?string $pref = null;
    public ?string $city = null;
    public ?string $address = null;
    public ?string $building = null;
    public ?string $tel = null;
    public ?string $email = null;
    public ?string $accessToken = null;
    public ?string $privateKey = null;
    public ?string $publicKey = null;
    public ?string $fingerprint = null;
    public ?ClientStatus $status = null;
    public ?Carbon $startAt = null;
    public ?Carbon $stopAt = null;

    /**
     * 一覧検索用キーワード（名前・識別子の部分一致）。API の keyword からマッピングします。
     */
    public ?string $keyword = null;

    /**
     * 利用開始日 From（Y-m-d 相当、空は無条件）。
     */
    public ?Carbon $startFrom = null;

    /**
     * 利用開始日 To（Y-m-d 相当、空は無条件）。
     */
    public ?Carbon $startTo = null;

    /**
     * 状態コードの一覧（空は無条件）。API の status クエリを正規化して設定します。
     *
     * @var array<int, int>
     */
    public array $statuses = [];
}

File: backends/java-spring/src/main/java/com/authorization/domain/client/condition/ClientCondition.java
/**
 * This is a program developed by BobTabo.
 *
 * Copyright (c) 2026 BobTabo. All Rights Reserved.
 */
package com.authorization.domain.client.condition;

import com.authorization.domain.client.enums.ClientStatus;
import com.authorization.support.repositories.conditions.AbstractCondition;
import java.time.LocalDateTime;
import java.util.List;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * クライアントConditionクラスです。
 *
 * @author Satoshi Nagashiba <satoshi.nagashiba@gmail.com>
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class ClientCondition extends AbstractCondition {

    private String name;
    private String identifier;
    private String postCode;
    private String pref;
    private String city;
    private String address;
    private String building;
    private String tel;
    private String email;
    private String accessToken;
    private String privateKey;
    private String publicKey;
    private String fingerprint;
    private ClientStatus status;
    private LocalDateTime startAt;
    private LocalDateTime stopAt;

    /** 一覧検索用キーワード（名前・識別子の部分一致）。 */
    private String keyword;

    /** 利用開始日 From（空は無条件）。 */
    private LocalDateTime startFrom;

    /** 利用開始日 To（空は無条件）。 */
    private LocalDateTime startTo;

    /** 状態コードの一覧（空は無条件）。 */
    private List<Integer> statuses = List.of();
}
